aiida/restapi/common/identifiers.py

```diff
@@ -103,16 +103,21 @@ def get_full_type_filters(full_type):
         node_type = escape_for_sql_like(node_type) + LIKE_OPERATOR_CHARACTER
         filters['node_type'] = {'like': node_type}
     else:
-        filters['node_type'] = escape_for_sql_like(node_type)
+        filters['node_type'] = node_type
```

Comment:
Moreover, I decided to remove the escaping when we were not using like because it would seem this is only for the like sql search?

```
def escape_for_sql_like(string):
    """Function that escapes % or _ symbols provided by user

    SQL LIKE syntax summary:

    - ``%`` -> match any number of characters
    - ``_`` -> match exactly one character

    """
    return string.replace('%', '\\%').replace('_', '\\_')
```

Moreover in testing this seems to be the case:

```
In [X]: qb = QueryBuilder()
In [X]: qb.append( orm.ProcessNode, filters={
In [X]:     'process_type': {'like': 'aiida\\_lsmo.workchains.isotherm.get\\_atomic\\_radii'}
In [X]:      })
In [X]: qb.count()
Out[X]: 348

In [X]: qb = QueryBuilder()
In [X]: qb.append( orm.ProcessNode, filters={
In [X]:     'process_type': 'aiida\\_lsmo.workchains.isotherm.get\\_atomic\\_radii'
In [X]:      })
In [X]: qb.count()
Out[X]: 0

In [X]: qb = QueryBuilder()
In [X]: qb.append( orm.ProcessNode, filters={
In [X]:     'process_type': 'aiida_lsmo.workchains.isotherm.get_atomic_radii'
In [X]:      })
In [X]: qb.count()
Out[X]: 348
```

Were we escaping this for any particular reason or was this a bug? @giovannipizzi @sphuber